### DISCUSSION TITLE
Need help with indeterminate offset error

### DISCUSSION THREAD

**Discussion by Techcable** [time-rs/time#455] [Question] [answered]
Hello! I am a co-maintainer for the `slog-term` crate, and I need help debugging [a time-related PR](https://github.com/slog-rs/term/pull/44).

Unfortunately fetching the local time seems broken, regardless of whether I do it inside or outside my application 😦

I am getting a "indeterminate offset" error on my local computer (M1 Mac), but it's also happening on Github Actions (Ubuntu).

Here is the old code (incorrectly uses UTC):
> let now: time::OffsetDateTime = std::time::SystemTime::now().into();

Here is the new code using local time that errors:
> let now = time::OffsetDateTime::now_local()
>        .map_err(convert_time_indeterminate_offset_error)?;

Here is the error:
> thread 'tests::test_logger' panicked at 'slog::Fuse Drain: Custom { kind: Other, error: IndeterminateOffset }', /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/slog-2.7.0/src/lib.rs:1944:33
[97](https://github.com/slog-rs/term/runs/5354381280?check_suite_focus=true#step:4:97)

I have the same one on my local machine.

I know very little about computer time, so I would really appreciate any help here :)

**Comment by jhpratt** [comment, marked as answer]
The method will fail on Unix systems when multiple threads are running. This is because it is otherwise unsound. For libraries, there is no workaround because you have no way of knowing what the end user will do with regard to threads.

**Reply by Techcable** [reply]
Thank you, you've saved me a good deal of confusion.
